application/blueprints/datamanager/services/async_api.py

```
@@ -79,21 +81,35 @@ def fetch_request(request_id: str) -> dict:
     return response.json() or {}
 
 
-def fetch_response_details(request_id: str, limit: int = 50) -> list:
+@cache.memoize(timeout=3600)
+def fetch_response_details(
+    request_id: str,
+    limit: int = 100,
+    start_offset: int = 0,
+    max_rows: int = None,
```

Ruff (0.15.12)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)